### DISCUSSION TITLE
M1 Pro, BUILD FAILED (macOS 12.0.1 using ruby-build 20211227) for 2.7.5

### DISCUSSION THREAD

**Discussion by slimdave** [rbenv/ruby-build#1888] [Build failures] [answered]
Currently on 2.7.2, using asdf ...

```
davidaldridge@Davids-MacBook-Pro consonance % asdf install ruby 2.7.5                 
Downloading openssl-1.1.1l.tar.gz...
-> https://dqw8nmjcqpjn7.cloudfront.net/0b7a3e5e59c34827fe0c3a74b7ec8baef302b98fa80088d7f9153aa16fa76bd1
Installing openssl-1.1.1l...
Installed openssl-1.1.1l to /Users/davidaldridge/.asdf/installs/ruby/2.7.5

Downloading ruby-2.7.5.tar.bz2...
-> https://cache.ruby-lang.org/pub/ruby/2.7/ruby-2.7.5.tar.bz2
Installing ruby-2.7.5...
ruby-build: using readline from homebrew

BUILD FAILED (macOS 12.0.1 using ruby-build 20211227)

Inspect or clean up the working tree at /var/folders/2d/4hfsnzp92rl90110xn8kh2tc0000gn/T/ruby-build.20220103113438.60145.tBsbRf
Results logged to /var/folders/2d/4hfsnzp92rl90110xn8kh2tc0000gn/T/ruby-build.20220103113438.60145.log

Last 10 log lines:
1 error generated.
compiling gc.c
make: *** [file.o] Error 1
make: *** Waiting for unfinished jobs....
In file included from gc.c:41:
./mjit.h:131:17: warning: cast to smaller integer type 'enum rb_mjit_iseq_func' from 'mjit_func_t' (aka 'unsigned long (*)(struct rb_execution_context_struct *, struct rb_control_frame_struct *)') [-Wpointer-to-enum-cast]
        switch ((enum rb_mjit_iseq_func)func) {
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 warning generated.
1 warning generated.
```

Log file gist here: https://gist.github.com/slimdave/1eaaa6c3ad7578fbc6fe8270d9b322d7

The relevant part of the log file seems to be ...
```
file.c:31:10: fatal error: 'CoreFoundation/CFString.h' file not found
#include <CoreFoundation/CFString.h>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from eval.c:20:
```
If relevant, which brew is `/opt/homebrew/bin/brew`.

Bit baffled, I'm afraid.

**Comment by slimdave** [comment]
Same for 2.7.4 and 2.7.3 FWIW.

I am pretty sure that the current 2.7.2 was migrated over from an M1 Mac Mini, which was running 12.0.0 at the time of installation, if that helps.

I've tried disabling Bitdefender, based on comments in other discussions, but no change.

**Comment by slimdave** [comment, marked as answer]
A pretty basic resolution to this - Xcode was not installed.

I'm assuming that the new Mac "inherited" compiled Ruby from the old one via the Mac Migration Assistant, but not Xcode, and installing was all that was required.

**Comment by TorvaldsDB** [comment]
Hi @slimdave how do you fix it? I'm still suffering the problem deeply.

**Reply by slimdave** [reply]
@TorvaldsDB I reinstalled Xcode and then installed Ruby. Did you migrate from a different laptop?

**Reply by TorvaldsDB** [reply]
Nope, it's my new laptop. 

by the way, how do you reinstall Xcode?

**Reply by slimdave** [reply]
I just did an Xcode install.

**Reply by TorvaldsDB** [reply]
thanks let me have a try first.

**Reply by TorvaldsDB** [reply]
Hi @silenius I fixed it. My problem is caused by the openssl. The `openssl@3.0` seems not compatible with the ruby-build. I switch it to `openssl@1.1` and it works.

Here is how I fixed it in detail https://github.com/rbenv/ruby-build/discussions/1853#discussioncomment-2146106 If someone meets a similar problem and finds it here, hope it will help you.

**Reply by lemeo** [reply]
[How to update Xcode from command line](https://stackoverflow.com/questions/34617452/how-to-update-xcode-from-command-line)

**Comment by Protino** [comment]
If someone is having issues related to block.h, simply remove / move it - https://github.com/rogual/neovim-dot-app/issues/331. I was able to install older versions (2.7.x)
